src/modules/job-manager/plugins/perilog.c

```diff
@@ -275,21 +285,27 @@ static bool perilog_log_ignore (struct perilog_conf *conf, const char *s)
     return false;
 }
 
-static void io_cb (flux_subprocess_t *sp, const char *stream)
+static void io_cb (struct bulk_exec *bulk_exec,
+                   flux_subprocess_t *sp,
+                   const char *stream,
+                   const char *data,
+                   int len,
+                   void *arg)
 {
-    struct perilog_proc *proc = flux_subprocess_aux_get (sp, "perilog_proc");
-    flux_t *h = flux_jobtap_get_flux (proc->p);
-    const char *s;
-    int len;
-
-    if ((len = flux_subprocess_getline (sp, stream, &s)) < 0) {
-        flux_log_error (h, "%s: %s: %s: flux_subprocess_getline",
-                        idf58 (proc->id),
-                        perilog_proc_name (proc),
-                        stream);
+    flux_t *h;
+    struct perilog_proc *proc;
+    char buf[len+1]; /* bulk_exec output is not NUL terminated */
+
+    if (!len
+        || !(proc = bulk_exec_aux_get (bulk_exec, "perilog_proc"))
+        || !(h = flux_jobtap_get_flux (proc->p)))
         return;
-    }
-    if (len && !perilog_log_ignore (&perilog_config, s)) {
+
+    /* Copy data to NUL terminated buffer */
+    memcpy (buf, data, len);
+    buf [len] = '\0';
```

Comment:
It is only passed to the log function after being checked if the line matches (or doesn't match) one of the log-ignore regexes. Since this function requires a NUL terminated buffer, it has to be copied.